phpBB/includes/acp/acp_extensions.php

```diff
@@ -145,14 +145,14 @@ function main()
 			break;
 
 			case 'enable_pre':
-				if (!$md_manager->validate_dir())
+				try
 				{
-					trigger_error($user->lang['EXTENSION_DIR_INVALID'] . adm_back_link($this->u_action), E_USER_WARNING);
+					$md_manager->validate_enable();
 				}
-
-				if (!$md_manager->validate_enable())
+				catch (exception_interface $e)
```

Comment:
I think is better to use \phpbb\extension\exception, and then add another catch with extension_interface that prints a string instead of a language key to allow extensions to launch their own custom error messages (compatible with cli and acp) and avoid this
https://github.com/david63/david63-sitelogo/blob/3.2/ext.php#L37
that is incompatible with cli.